tests/drivers/spi/spi_slave_master_loopback/src/spi.c

```diff
@@ -0,0 +1,418 @@
+/*
+ * Copyright (c) 2017 BayLibre, SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_LEVEL SYS_LOG_LEVEL_DEBUG
+#include <logging/sys_log.h>
+
+#include <zephyr.h>
+#include <misc/printk.h>
+#include <string.h>
+#include <stdio.h>
+
+#include <spi.h>
+
+#ifdef CONFIG_TEST_SPI_MASTER_CS_GPIO_ENABLE
+struct spi_cs_control spi_master_cs = {
+	.gpio_pin = 0,
+	.delay = 0
+};
+#define SPI_MASTER_CS (&spi_master_cs)
+#else
+#define SPI_MASTER_CS NULL
+#endif
+
+#ifdef CONFIG_TEST_SPI_SLAVE_CS_IGNORE
+/* Ignore CS for slave */
+struct spi_cs_control spi_slave_cs = {
+	.gpio_pin = 0,
+	.delay = 0
+};
+#define SPI_SLAVE_CS (&spi_slave_cs)
```

Comment:
Sorry, I'm confused about this behavior, probably I am missing something.

If you want to ignore CS on the slave (i.e. CONFIG_TEST_SPI_SLAVE_CS_IGNORE=y), shouldn't SPI_SLAVE_CS be NULL here, instead of &spi_slave_cs? I.e. shouldn't this be the code for the `#else` branch?

Reply:
Yes, I forgot it, I'll add it asap

Reply:
OK, It needs a comment here to explain.

On STM32, there is 3 behaviours:
- cs=NULL leads to use the Hardware NSS pin, if not muxed it will never start
- cs=valid, dev=NULL leads to ignore Hardware NSS and won't use a GPIO
- cs=valid, dev=valid leads to use a GPIO as CS, but it makes no sense (yet) in slave mode

With @tbursztyka we aggred about that and the spi_context.h handles the second case.

I added a comment for that